lametro/events.py

```diff
@@ -12,13 +12,106 @@ class LametroEventScraper(LegistarAPIEventScraper):
     EVENTSPAGE = "https://metro.legistar.com/Calendar.aspx"
     TIMEZONE = "America/Los_Angeles"
 
+    def _pair_events(self, events):
+        paired_events = []
+        unpaired_events = []
+
+        for incoming_event in events:
+            try:
+                partner_event, = [e for e in unpaired_events
+                                  if incoming_event.is_partner(e)]
+            except ValueError:
+                unpaired_events.append(incoming_event)
+
+            else:
+                unpaired_events.remove(partner_event)
+                paired_events.append(incoming_event)
+                paired_events.append(partner_event)
+
+        return paired_events, unpaired_events
+
+    def _find_partner(self, event):
+
+        results = list(self.search('/events/', 'EventId',
+                                   event.partner_search_string))
+        if results:
+            partner, = results
+            partner = LAMetroAPIEvent(partner)
+            assert event.is_partner(partner)
+            return partner
+
+        elif event.is_spanish:
+            raise ValueError("Can't find English companion for Spanish Event {}".format(event['EventId']))
+
+        else:
+            return None
+        
+
+    def api_events(self, *args, **kwargs):
+        '''
+        For meetings, Metro provides an English audio recording and a
+        Spanish recording. Due to limitations with the InSite system,
+        multiple audio recordings can't be associated with a single
+        InSite event. So, Metro creates two InSite event entries for
+        the same actual event, one Insite event entry has the English
+        audio and the other has the Spanish Audio. The Spanish InSite
+        event entry has the same name as the English event entry,
+        except the name is suffixed with ' (SAP)'.
+
+        We need to merge these companion events. In order to do that,
+        we must ensure that if we scrape one member of pair, we also
+        scrape its partner.
+
+        This method subclasses the normal api_event method to ensure
+        that we get both members of pairs.
+        '''
+        events = (LAMetroAPIEvent(event) for event
+                  in super().api_events(*args, **kwargs))
+
+        paired, unpaired = self._pair_events(events)
+
+        yield from paired
+
+        for unpaired_event in unpaired:
+            yield unpaired_event
+
+            partner_event = self._find_partner(unpaired_event)
+            if partner_event is not None:
```

Comment:
I'd add a comment here, or a docstring in `_find_partner` that says we'll allow unpartnered English events.

Reply:
Added docstring to find_partner